packages/DateInput/src/DateInput.js

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+import PropTypes from "prop-types";
+import moment from "moment";
+import CalendarIcon from "@paprika/icon/lib/Calendar";
+import Input from "@paprika/input";
+import { isMomentObjectOrNull } from "@paprika/helpers/lib/customPropTypes";
+import useI18n from "@paprika/l10n/lib/useI18n";
+
+const INPUT_PARSE_ERROR = "INPUT_PARSE";
+
+const propTypes = {
+  /** If the value of <input> is valid or not. */
+  hasError: PropTypes.bool,
+
+  /** Date format used while entering and parsing user input. */
+  dateFormat: PropTypes.string,
+
+  /** Selected date in moment object. */
+  date: isMomentObjectOrNull,
+
+  /** Possible date in moment object. This prop is used when user is typing date and we can't determinate exact date. */
+  possibleDate: isMomentObjectOrNull,
```

Comment:
I'm a little confused, why would you want to pass-down a `possibleDate` to the component shouldn't this be something like onPossibleDate callback, like what you would receive after changing the date prop? 

Also **NO** for this PR but would be cool if we could have a `useDateInput` hook in some form of:

```
import { useDateInput } from "@paprika/date-input"
const { date, possibleDate } = useDateInput(defaultDate);
```
we should provide this uncontrolled option for all our components, our regular input would benefit having a similar pattern as well. 

cc @mikrotron @tristanjasper

Reply:
I suspect this was just moved over from the previous version.  I remember @allison-c explaining this to me when she initially added it, and it was important for the formatting, but I don't remember the exact details.  I don't think the purpose of this PR is to revisit the working logic though, so it might be best to leave it as is?

Reply:
@nahumzs 
> why would you want to pass-down a possibleDate to the component 

Oh, thanks! I just realized I do not use this prop inside DateInput. Will remove it and leave here only `onChangePossibleDate` callback.